.github/workflows/python_push.yml

```diff
@@ -0,0 +1,62 @@
+name: python build
+
+on:
+  pull_request:
+    types: [opened, synchronize]
+    paths:
+      - experimental/python/**
+  merge_group:
+    types: [checks_requested]
+    paths:
+      - experimental/python/**
+  push:
+    # Always run on push to main. The build cache can only be reused
+    # if it was saved by a run from the repository's default branch.
+    # The run result will be identical to that from the merge queue
+    # because the commit is identical, yet we need to perform it to
+    # seed the build cache.
+    branches:
+      - main
+
+jobs:
+  python_tests:
+    name: tests
+    runs-on: ubuntu-latest
```

Comment:
why no OS in the matrix?

Reply:
From PR description:
> These jobs are separate from Golang code because they are executed against a matrix of Python versions, and OS versions are not relevant. We will get coverage with OS versions using acceptance tests. In addition, we don't need to run these tests unless any of the Python code is modified.